File: Java Labs/Patterns On Java/src/com/pattern/decorator/SizeDecorator.java
package com.pattern.decorator;

public class SizeDecorator extends ClothingItemDecorator {
    private int size;

    public SizeDecorator(ClothingItem decoratedItem, int size) {
        super(decoratedItem);
        this.size = size;
    }

    @Override
    public String getDescription() {
        return super.getDescription() + " Size: " + size;
    }

    @Override
    public double getPrice() {
        // Additional cost for size
        return super.getPrice() + 10.0;
    }
}

File: Python Labs/com/pattern/decorator/SizeDecorator.py
from com.pattern.decorator.ClothingItemDecorator import ClothingItemDecorator


class SizeDecorator(ClothingItemDecorator):
    def __init__(self, decorated_item, size):
        super().__init__(decorated_item)
        self.size = size

    def get_description(self):
        return super().get_description() + f" Size: {self.size}"

    def get_price(self):
        return super().get_price() + 10.0  # Additional cost for size